libs/server/core/application-services/src/lib/activity-post/activity-post.service.ts

```diff
@@ -98,4 +109,31 @@ export class ActivityPostService {
         const currentPost = await this.activityPostRepo.findOneOrFail(dto.postId, ActivityPostQuery);
         return this.activityPostRepo.update(dto.postId, {likeCount: currentPost.likeCount - 1}, query);
     }
+
+    async digest(query: IQuery<ActivityPost[]>, token: string, dto: DigestActivityPostDto) {
+        const user = await this.auth.validateUserToken(token ?? '');
+        const startDate = new Date(dto.startDate);
+        const posts = await this.activityPostRepo.query(query, { where: { user: user.id }}); // todo -> way to query for dates here instead??
+        const res: IParserObject<ActivityPost, IQueryObject<ActivityPost> & IPaginate>[] = [];
+        posts.forEach(post => {
+            post.comments = post.comments?.filter(comment => new Date(comment?.dateCreated as any) >= startDate);
+            post.likes = post.likes?.filter(like => new Date(like?.dateCreated as any) >= startDate);
+            if ((post.comments?.length ?? 0 > 0) || (post.likes?.length ?? 0 > 0)) res.push(post);
+        });
+        return res;
+    }
+    // @Cron("0 20 * * 2") every tuesday at 8pm
+    @Cron(CronExpression.EVERY_MINUTE)
```

Comment:
@AlexanderChiuy Confused on when this job is used, every minute??

Reply:
It supposed to be every Tuesday eat 8pm but I left every minute for testing purposes. Its fixed now